### PR TITLE
Internal: Editor V2 - Adjust eslint [ED-9246]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: packages/.eslintrc.js [hunk 1]
```diff
@@ -16,14 +16,25 @@ module.exports = {
 		},
 	},
 	rules: {
+		// ignore when trying to use '@elementor/*' packages
 		'import/no-unresolved': [ 'error', { ignore: [ '@elementor/.+' ] } ],
 
-		// Disable the js no-unused-vars rule, and enable the ts version.
-		'no-unused-vars': 'off',
-		'@typescript-eslint/no-unused-vars': [ 'error' ],
+		// Don't allow relative import from package to package
```
Comment (StyleShit) [inline packages/.eslintrc.js:22]:
```suggestion
		// Don't allow relative import from package to package.
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: packages/.eslintrc.js [hunk 2]
```diff
@@ -16,14 +16,25 @@ module.exports = {
 		},
 	},
 	rules: {
+		// ignore when trying to use '@elementor/*' packages
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/.eslintrc.js:19]: Done

File: packages/.eslintrc.js [hunk 3]
```diff
@@ -16,14 +16,25 @@ module.exports = {
...
+			{
+				selector: 'ImportDeclaration[source.value=/^@elementor\\u002F.+\\u002F/]',
+				message: 'Path access on Elementor dependencies is not allowed, please access using only root level, e.g: "@elementor/locations" and not "@elementor/locations/src/index.ts".',
+			},
+		],
 
 		// Disable import/named rule, TS will handle it.
 		'import/named': [ 'off' ],
+
+		// Disable the js no-unused-vars rule, and enable the ts version.
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/.eslintrc.js:35]: Done

File: packages/.eslintrc.js [hunk 4]
```diff
@@ -16,14 +16,25 @@ module.exports = {
...
-		// Disable the js no-unused-vars rule, and enable the ts version.
-		'no-unused-vars': 'off',
-		'@typescript-eslint/no-unused-vars': [ 'error' ],
+		// Don't allow relative import from package to package
+		'import/no-relative-packages': [ 'error' ],
+		'no-restricted-syntax': [
+			'error',
+			{
+				selector: 'ImportDeclaration[source.value=/^@elementor\\u002F.+\\u002F/]',
+				message: 'Path access on Elementor dependencies is not allowed, please access using only root level, e.g: "@elementor/locations" and not "@elementor/locations/src/index.ts".',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/.eslintrc.js]: Done

File: packages/.eslintrc.js (hunk 1 above)
Comment (Nevoss) [reply packages/.eslintrc.js]: Done

**Review by StyleShit — APPROVED** [review]

**Pull request merged**